enlighten/SPCFileWriter.py

```diff
@@ -0,0 +1,672 @@
+
+def fit_byte_block(field: bytearray, limit: int) -> bytes:
+    while len(field) < limit:
+        field.extend(bytearray(b"\x00"))
+    if len(field) > limit:
+        field = field[:limit]
+        field[-1] = 0 # most fields are null terminated so just make that the default
+    return bytes(field)
+
+class SPCFileType(IntFlag):
+    """
+    Describes the various file format flags.
+    See the old specification for greater detail on what each setting changes.
+    """
+    DEFAULT = 0 # Will be 32 bit single file, one trace, with even x values
+    SIXTEENPREC = 1	# 32 bit float or 16 bit spc format, currently only support 32 
+    TCGRAM = 2	# Enables fexper in older software (not used accroding to old format doc)
+    TMULTI = 4	# Multiple traces format (set if more than one subfile), Y vs YYYY, XY vs XYYY/XYXY
+    TRANDM = 8	# If TMULTI and TRANDM=1 then arbitrary time (Z) values, must pass full z array
+    TORDRD = 16	# If TMULTI and TORDRD=1 then ordered but uneven z, must pass full z array if true, else only z_inc and single z value in z array 
+    TALABS = 32	# If true, use the specified custom axes
+    TXYXYS = 64	# If true the file is XYXYXY, flase is considered XYYYYYY 
+    TXVALS = 128	# Non even X values, must pass full x array 
+
+# From GRAMSDDE.h
+# Currenlty only support PPNONE
+class SPCProcessCode(IntEnum):
+    PPNONE	= 0    # No post processing 
+    PPCOMP	= 1    # Compute (run PPCOMP?.ABP) 
+    PPDLLC	= 2    # Compute with DLL (run PPCOMP?.DLL) 
+    PPTRANS = 4    # Transmission (run PPTRANS?.ABP) 
+    PPABSRB = 8    # Absorbance (run PPABSRB?.ABP) 
+    PPKMUNK = 12  # Kuebelka-Munk (run PPKMUNK?.ABP) 
+    PPPEAK	= 32   # GRAMS built-in peak picking and reporting 
+    PPSRCH	= 64   # Library Search associated w/experiment's LIB driver 
+    PPUSER	= 128  # General user-written post program (run PPUSER?.ABP) 
+
+class SPCModFlags(IntFlag):
+    Not = 0 # unmodified
+    A = 2**1 # Averaging (from multiple source traces)
+    B = 2**2 # Baseline correction or offset functions
+    C = 2**3 # Interferogram to spectrum Computation
+    D = 2**4 # Derivative (or integrate) functions
+    E = 2**6 # Resolution Enhancement functions (such as deconvolution)
+    I = 2**9 # Interpolation functions
+    N = 2**14 # Noise reduction smoothing
+    O = 2**15 # Other functions (add, subtract, noise, etc.)
+    S = 2**19 # Spectral Subtraction
+    T = 2**20 # Truncation (only a portion of original X axis remains)
+    W = 2**20 # When collected (date and time information) has been modified
+    X = 2**24 # X units conversions or X shifting
+    Y = 2**25 # Y units conversions (transmission->absorbance, etc.)
+    Z = 2**26 # Zap functions (features removed or modified)
+
+# looking at example parsers,
+# axes z and w use the same units
+class SPCXType(IntEnum):
+    SPCXArb	        = 0
+    SPCXWaven	    = 1
+    SPCXUMetr	    = 2
+    SPCXNMetr	    = 3
+    SPCXSecs	    = 4
+    SPCXMinuts	    = 5
```

Comment:
I know you didn't do this but I look at "Metr" and "Minuts" and wonder if the designer had a broken keyboard